src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,452 @@
+    [UnixCompatible]
+    public class HPLinpackExecutor : VirtualClientComponent
+    {
+        private IFileSystem fileSystem;
+        private ISystemManagement systemManagement;
+        private IStateManager stateManager;
+        private IPackageManager packageManager;
+
+        /// <summary>
+        /// Constructor for <see cref="HPLinpackExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public HPLinpackExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+             : base(dependencies, parameters)
+        {
+            this.systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            this.packageManager = this.systemManagement.PackageManager;
+            this.fileSystem = this.systemManagement.FileSystem;
+            this.stateManager = this.systemManagement.StateManager;
+        }
+
+        /// <summary>
+        /// The order of the coefficient matrix also known as problem size (N)
+        /// </summary>
+        public string N
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(HPLinpackExecutor.N), out IConvertible n);
+                return n?.ToString();
+            }
+
+            set
+            {
+                this.Parameters[nameof(this.N)] = value;
+            }
+        }
+
+        /// <summary>
+        /// The partitioning blocking factor also known as block size (NB)
+        /// </summary>
+        public string NB
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(HPLinpackExecutor.NB), out IConvertible nb);
+                return nb?.ToString();
+            }
+
+            set
+            {
+                this.Parameters[nameof(this.NB)] = value;
+            }
+        }
+
+        /// <summary>
+        /// The name of the package where the ARMPerformanceLibraries package is downloaded.
+        /// </summary>
+        public string ARMPerformanceLibrariesPackageName
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(HPLinpackExecutor.ARMPerformanceLibrariesPackageName), out IConvertible armperformancelibraries);
+                return armperformancelibraries?.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Version of HPL being used
+        /// </summary>
+        public string HPLVersion
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(HPLinpackExecutor.HPLVersion), out IConvertible hplversion);
+                return hplversion?.ToString();
+            }
+        }
+
+        /// <summary>
+        /// The path where the HPL directory.
+        /// </summary>
+        public string HPLDirectory { get; set; }
```

Comment:
Properties that are not parameter readers should be protected or private member variables of the class.